src/Symfony/Component/Translation/Formatter/MessageFormatterInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Formatter;
+
+/**
+ * MessageFormatterInterface.
+ *
+ * @author Guilherme Blanco <guilhermeblanco@hotmail.com>
+ */
+interface MessageFormatterInterface
+{
+    /**
+     * Formats a lozalized message pattern with given arguments.
+     *
+     * @param string   $locale    The message locale
+     * @param string   $id        The message id (may also be an object that can be cast to string)
+     * @param int|null $number    The number to use to find the indice of the message (if exists)
```

Comment:
@fabpot I'm not sure if it's a good idea,  I would remove `number` argument If we planned to use [IntlMessageFormatter](https://github.com/aitboudad/symfony/blob/translation_intl_formatter/src/Symfony/Component/Translation/Formatter/IntlMessageFormatter.php) as default formatter in `3.0` and I think we don't need anymore [transChoice](https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/Translation/Translator.php#L237)

So what do you think about deprecation `transChoice` method and [MessageSelector](https://github.com/aitboudad/symfony/blob/translation_intl_formatter/src/Symfony/Component/Translation/MessageSelector.php) then the interface become :

```php
interface MessageFormatterInterface
{
    public function format($locale, $id, array $arguments = array());
}
```

if not any idea of the name of methods in the interface ?